pkg/engines/slurm/slurm.go

```diff
@@ -148,8 +148,17 @@ func GenerateOutputParams(ctx context.Context, tree *topology.Vertex, params *Pa
 			plugin = tree.Metadata[topology.KeyPlugin]
 		}
 		if len(plugin) == 0 {
+			if tree.Vertices[topology.TopologyBlock] != nil {
+				plugin = topology.TopologyBlock
+			} else {
+				plugin = topology.TopologyTree
+			}
+		} else if plugin == topology.TopologyBlock && tree.Vertices[topology.TopologyBlock] == nil {
+			klog.Infof("Admin provided plugin %v not valid, overriding with %v", plugin, topology.TopologyTree)
```

Comment:
please add a comment for future task:
"//TODO: add Prometheus metrics to surface this error"
We need to track and report all errors.

Reply:
Done.